ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -737,7 +737,7 @@ static char *make_filename_safe(const char *filename)
 		}
 		return estrdup(filename);
 	}
-	if (*filename && strcmp(filename, ":memory:")) {
+	if (*filename && strncmp(filename, ":memory:", sizeof(":memory:")-1) != 0) {
```

Comment:
Is this 100% safe and wouldn't be something like `:memory:/path` processed like regular filepath?

Reply:
The official condition is in https://github.com/sqlite/sqlite/blob/version-3.50.4/src/btree.c#L2552.

I even wonder what is `:memory:?cache=shared`, I did not find any testcase for it in Sqlite codebase.

Reply:
The reason why this does not work is in https://www.sqlite.org/sharedcache.html#shared_cache_and_in_memory_databases.